src/tasks/netSdk/NetSdkRunTaskProvider.ts

```diff
@@ -3,56 +3,91 @@
  *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import * as path from "path";
 import { CancellationToken, CustomExecution, Task, TaskDefinition, TaskScope } from "vscode";
 import { DockerPseudoterminal } from "../DockerPseudoterminal";
+import { DockerRunTask } from "../DockerRunTaskProvider";
 import { DockerTaskProvider } from '../DockerTaskProvider';
-import { DockerTaskExecutionContext } from '../TaskHelper';
+import { DockerRunTaskContext } from '../TaskHelper';
+import { NetCoreRunTaskDefinition, NetCoreTaskHelper } from "../netcore/NetCoreTaskHelper";
 import { NetSdkRunTaskType, getNetSdkBuildCommand, getNetSdkRunCommand } from './netSdkTaskUtils';
 
 const NetSdkDebugTaskName = 'debug';
+
+export type NetSdkRunTask = NetCoreRunTaskDefinition;
+
 export class NetSdkRunTaskProvider extends DockerTaskProvider {
 
     public constructor() { super(NetSdkRunTaskType, undefined); }
 
     public provideTasks(token: CancellationToken): Task[] {
-
-        // we need to initialize a task first so we can pass it into `DockerPseudoterminal`
-        const task = new Task(
-            { type: NetSdkRunTaskType },
-            TaskScope.Workspace,
-            NetSdkDebugTaskName,
-            NetSdkRunTaskType
-        );
-
-        task.execution = new CustomExecution(async (resolvedDefinition: TaskDefinition) =>
-            Promise.resolve(new DockerPseudoterminal(this, task, resolvedDefinition))
-        );
-
-        return [task];
+        return [this.createNetSdkRunTask().task];
```

Comment:
Thinking about the new flow, we can probably just return [] here for now, since the ambient debug task should handle actually invoking build/run, so we shouldn't need to actually provide an ambient build task.

Reply:
that makes sense to me, let me do some testing

Reply:
test results: the method does not get called (breakpoint is not hit) and therefore we should just change it to `return []` and add a comment like `// this task is not discoverable this way`